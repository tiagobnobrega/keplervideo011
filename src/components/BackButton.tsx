import MaterialIcons from '@amzn/react-native-vector-icons/MaterialIcons';
import { isEqual } from 'lodash';
import React from 'react';
import { StyleProp, StyleSheet, Text, ViewStyle } from 'react-native';
import { COLORS } from '../styles/Colors';
import { scaleUxToDp } from '../utils/pixelUtils';
import FocusableElement from './FocusableElement';

interface BackButtonProps {
  onPress: () => void;
  overrideStyle?: StyleProp<ViewStyle>;
  hasTVPreferredFocus: boolean;
}

const BackButton = ({
  onPress,
  overrideStyle,
  hasTVPreferredFocus,
}: BackButtonProps) => {
  return (
    <FocusableElement
      hasTVPreferredFocus={hasTVPreferredFocus}
      style={[styles.backButtonContainer, overrideStyle]}
      onFocusOverrideStyle={styles.backButtonFocus}
      onPress={onPress}>
      <Text testID={'header_back_icon'}>
        <MaterialIcons name={'chevron-left'} size={70} color={COLORS.WHITE} />
      </Text>
    </FocusableElement>
  );
};

const backButtonDataUnchanged = (
  prevProps: BackButtonProps,
  nextProps: BackButtonProps,
) => {
  return isEqual(prevProps, nextProps);
};

export default React.memo(BackButton, backButtonDataUnchanged);

const styles = StyleSheet.create({
  backButtonImage: {
    height: scaleUxToDp(70),
    width: scaleUxToDp(70),
    resizeMode: 'contain',
    tintColor: COLORS.WHITE,
  },
  backButtonContainer: {
    padding: scaleUxToDp(25),
    alignItems: 'center',
    flexDirection: 'row',
    width: scaleUxToDp(125),
  },
  backButtonFocus: {
    backgroundColor: COLORS.DARK_GRAY,
    borderRadius: scaleUxToDp(70),
  },
});
