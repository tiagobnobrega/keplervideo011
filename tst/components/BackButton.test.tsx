import { render } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import BackButton from '../../src/components/BackButton';

jest.mock('../../src/components/FocusableElement');
const onPressMock = jest.fn();
describe('BackButton renders correctly', () => {
  it('renders correctly without styles', () => {
    const tree = render(
      <BackButton onPress={onPressMock} hasTVPreferredFocus={true} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
describe('BackButton with React.memo', () => {
  it('renders only when props change', async () => {
    const { rerender, UNSAFE_getByProps } = render(
      <BackButton onPress={onPressMock} hasTVPreferredFocus={true} />,
    );
    const isEqualSpy = jest.spyOn(require('lodash'), 'isEqual');
    rerender(<BackButton onPress={onPressMock} hasTVPreferredFocus={true} />);
    expect(isEqualSpy).toHaveBeenCalledWith(
      { hasTVPreferredFocus: true, onPress: onPressMock },
      { hasTVPreferredFocus: true, onPress: onPressMock },
    );
    rerender(<BackButton onPress={onPressMock} hasTVPreferredFocus={false} />);
    expect(isEqualSpy).toHaveBeenCalledWith(
      { hasTVPreferredFocus: true, onPress: onPressMock },
      { hasTVPreferredFocus: false, onPress: onPressMock },
    );
    expect(UNSAFE_getByProps({ hasTVPreferredFocus: false })).toBeTruthy();
  });
});
