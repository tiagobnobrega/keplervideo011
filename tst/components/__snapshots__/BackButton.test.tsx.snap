// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly renders correctly without styles 1`] = `null`;
